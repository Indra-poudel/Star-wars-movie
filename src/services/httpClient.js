import axios from 'axios';

import {STAR_WAR_BASE_URL} from '../constants/endpoints';

export const httpClient = axios.create({
  baseURL: STAR_WAR_BASE_URL,
  headers: {
    'Content-Type': 'application/json',
  },
});
