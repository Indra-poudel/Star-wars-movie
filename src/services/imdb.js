import {httpClient} from './httpClient';

import {GET_RATING, IMDB_BASE_URL} from '../constants/endpoints';
import {interpolate} from '../utils/string';
import {IMDB_API_KEY} from '../constants/apiKey';

export const getIMDbRating = async id => {
  const endpoint = interpolate(GET_RATING, {apiKey: IMDB_API_KEY, imdbId: id});

  const response = await httpClient.get(endpoint, {
    baseURL: IMDB_BASE_URL,
  });

  return response.data.imDb;
};
