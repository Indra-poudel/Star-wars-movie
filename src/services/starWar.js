import {httpClient} from './httpClient';
import {
  GET_ALL_START_WAR_EPISODES,
  GET_CHARACTER_INFO,
} from '../constants/endpoints';
import {interpolate} from '../utils/string';

export const getListOfStarWarEpisode = async () => {
  const response = await httpClient.get(GET_ALL_START_WAR_EPISODES);
  return response.data?.results || [];
};

export const getCharacterInfo = async id => {
  const url = interpolate(GET_CHARACTER_INFO, {characterId: id});
  const response = await httpClient.get(url);

  return {
    name: response.data.name,
    height: response.data.height,
    gender: response.data.gender,
  };
};
