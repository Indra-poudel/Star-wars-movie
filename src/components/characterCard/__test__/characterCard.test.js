import React from 'react';
import CharacterCard from '../characterCard';
import renderer from 'react-test-renderer';
import {httpClient} from '../../../services/httpClient';

jest.mock('axios', () => {
  return {
    create: jest.fn(() => ({
      post: jest.fn(),
      get: jest.fn(),
      interceptors: {
        request: {use: jest.fn(), eject: jest.fn()},
        response: {use: jest.fn(), eject: jest.fn()},
      },
    })),
  };
});

test('Verify Api data in the component', () => {
  const resp = {
    data: {name: 'Indra Poudel', height: '5.8', gender: 'male'},
  };

  httpClient.get.mockResolvedValue(resp);

  const tree = renderer
    .create(<CharacterCard url={'https://swapi.dev/api/people/1/'} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
