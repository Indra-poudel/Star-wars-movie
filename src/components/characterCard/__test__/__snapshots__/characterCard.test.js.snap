// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Verify Api data in the component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderColor": "black",
      "borderWidth": 1,
      "display": "flex",
      "elevation": 5,
      "flexDirection": "column",
      "height": 60,
      "justifyContent": "center",
      "margin": 8,
      "textAlign": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "black",
        "fontFamily": "ProximaNova-Regular",
        "fontSize": 18,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    Loading...
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontFamily": "ProximaNova-Regular",
          "fontSize": 14,
          "marginHorizontal": 12,
          "textAlign": "center",
        }
      }
    >
      Gender:
       
      Loading...
    </Text>
    <Text
      style={
        Object {
          "color": "black",
          "fontFamily": "ProximaNova-Regular",
          "fontSize": 14,
          "marginHorizontal": 12,
          "textAlign": "center",
        }
      }
    >
      |
    </Text>
    <Text
      style={
        Object {
          "color": "black",
          "fontFamily": "ProximaNova-Regular",
          "fontSize": 14,
          "marginHorizontal": 12,
          "textAlign": "center",
        }
      }
    >
      Height:
       
      Loading...
    </Text>
  </View>
</View>
`;
